src/index/ivf_raft/ivf_raft.cuh

```diff
@@ -37,6 +37,20 @@
 
 namespace knowhere {
 
+namespace memory_pool {
+static rmm::mr::device_memory_resource*
+resources() {
+    auto device_id = rmm::detail::current_device();
```

Comment:
I propose we not invoke anything in the `detail` namespace (for RMM or for RAFT). That namespace is subject to change without warning and could cause integration nightmares going forward.